freshmaker/handlers/koji/rebuild_flatpak_application_on_module_ready.py

```diff
@@ -20,13 +20,148 @@
+from freshmaker.models import Event
+from freshmaker.pyxis import Pyxis
+from freshmaker.types import EventState
 
 
 class RebuildFlatpakApplicationOnModuleReady(ContainerBuildHandler):
-    name = 'RebuildFlatpakApplicationOnModuleReady'
+    name = "RebuildFlatpakApplicationOnModuleReady"
 
     def can_handle(self, event):
         return isinstance(event, FlatpakModuleAdvisoryReadyEvent)
+
+    @fail_event_on_handler_exception
+    def handle(self, event):
+
+        if event.dry_run:
+            self.force_dry_run()
+
+        self.event = event
+
+        db_event = Event.get_or_create_from_event(db.session, event)
+        self.set_context(db_event)
+
+        auto_build_images_list = self._get_auto_rebuild_image_list(event)
+        rebuild_images = self._filter_images_with_higher_rpm_nvr(
+            event, auto_build_images_list
+        )
+        if not auto_build_images_list or not rebuild_images:
+            msg = "There is no image can be rebuilt. " f"message_id: {event.msg_id}"
+            db_event.transition(EventState.SKIPPED, msg)
+            db.session.commit()
+            self.log_info(msg)
+            return []
+
+        images_nvr = ",".join([rebuild_image.nvr for rebuild_image in rebuild_images])
+        self.log_info("Following images %s will be rebuilt" % images_nvr)
+
+        # TODO: Return empty list so far as this method is still in progress.
+        # Will think about whether to return the real BaseEvent objects in future.
+        return []
+
+    def _get_requests_session(cls, retry_options={}):
+        """
+        Create a requests session. Reference from:
+        https://github.com/release-engineering/cachito/blob/master/cachito/workers/requests.py
+
+        :param dict retry_options: overwrite options for initialization of Retry instance
+        :return: the configured requests session
+        :rtype: requests.Session
+        """
+        DEFAULT_RETRY_OPTIONS = {
+            "total": 5,
+            "read": 5,
+            "connect": 5,
+            "backoff_factor": 1.3,
+            "status_forcelist": (500, 502, 503, 504),
+        }
+        session = requests.Session()
+        retry_options = {**DEFAULT_RETRY_OPTIONS, **retry_options}
+        adapter = HTTPAdapter(max_retries=Retry(**retry_options))
+        session.mount("http://", adapter)
+        session.mount("https://", adapter)
+        return session
+
+    def _get_auto_rebuild_image_list(self, event):
+        """
+        Get rebuild image list which can be auto rebuilt.
+
+        :param event FlatpakModuleAdvisoryReadyEvent: The event this handler
+            is currently handling.
+        :rtype: list
+        :return: List of images which can be auto rebuilt.
+        """
+        if not conf.pyxis_server_url:
+            raise ValueError("'PYXIS_SERVER_URL' parameter should be set")
+        self._pyxis = Pyxis(conf.pyxis_server_url)
+
+        if not conf.flatpak_server_url or "://" not in conf.flatpak_server_url:
```

Comment:
What's the reason for checking the `://` in URL? I would rather not do any assumptions about format of the URL.

```suggestion
        if not conf.flatpak_server_url:
```

Reply:
Actually no strong opinion. Fixing.